modules/javafx.media/src/main/native/gstreamer/plugins/mfwrapper/mfwrapper.cpp

```diff
@@ -236,22 +238,33 @@ static void gst_mfwrapper_init(GstMFWrapper *decoder)
     decoder->defaultStride = 0;
     decoder->pixel_num = 0;
     decoder->pixel_den = 0;
+
+    decoder->set_caps = TRUE;
 }
 
 static void gst_mfwrapper_dispose(GObject* object)
 {
     GstMFWrapper *decoder = GST_MFWRAPPER(object);
 
-    // No need to free pDecoderBuffer, it will be release when interface is
-    // release by MF.
+    if (decoder->header != NULL)
+    {
+        delete[] decoder->header;
+        decoder->header = NULL;
+        decoder->header_size = 0;
+    }
+
+    // No need to free pDecoderBuffer, it will be release when
+    // pDecoderOutput is released.
     SafeRelease(&decoder->pDecoderOutput);
+    decoder->pDecoderBuffer = NULL;
```

Comment:
Is setting this to NULL sufficient or does it need to be freed / released first?